arraycontext/container/arithmetic.py

```diff
@@ -357,10 +373,18 @@ def {fname}(arg1):
                                 else:
                                     raise ValueError(msg)""")
                     gen(f"return cls({zip_init_args})")
+
+                if bcast_actx_array_type:
+                    all_outer_bcast_type_names = (
+                        outer_bcast_type_names
+                        + ("*arg1.array_context.get_array_types()",))
+                else:
+                    all_outer_bcast_type_names = outer_bcast_type_names
```

Comment:
* Move this closer to the usage site.
* Make this produce a tuple that's either empty or has the actx array type(s) in it. Add both together in the argument of `tup_str`.

(Same below for reverse.)

Reply:
Thanks, I think 2040a8d should fix it.